client/packages/sdk/src/price-estimation/__tests__/substrate.test.ts

```diff
@@ -0,0 +1,98 @@
+import { describe, expect, it } from "@jest/globals";
+import { createType } from "@t3rn/types"
+import { ApiPromise, Keyring, Sdk, cryptoWaitReady } from "../../index";
+import { Actions, Targets, estimateGasFee, estimateBidAmount, estimateMaxReward } from "../index";
+import { EstimateSubmittableExtrinsicParams } from "../substrate";
+
+const createCircuitContext = async (
+  exportMode = false,
+) => {
+  await cryptoWaitReady()
+  const keyring = new Keyring({ type: "sr25519" })
+  const signer =
+    process.env.CIRCUIT_KEY === undefined
+      ? keyring.addFromUri("//Alice")
+      : keyring.addFromMnemonic(process.env.CIRCUIT_KEY)
+  const sdk = new Sdk(
+    "ws://localhost:9944",
+    signer,
+    exportMode,
+  )
+  const circuit = await sdk.init()
+  return {
+    circuit,
+    sdk,
+    signer,
+  }
+}
+
+const buildSfx = (circuit: ApiPromise
+) => {
+  return {
+    sideEffects: createType("Vec<T3rnTypesSfxSideEffect>", [{
+      target: "roco",
+      maxReward: 1000000,
+      insurance: 1000000,
+      encodedArgs: ["0x0", "0x1"],
```

Comment:
I think its important to have proper arg length here, as tx bytes is part of the tx cost